packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/useTimePicker.tsx

```diff
@@ -70,12 +70,12 @@ export const useTimePicker_unstable = (props: TimePickerProps, ref: React.Ref<HT
     initialState: undefined,
   });
 
-  const selectedTimeTextRef = React.useRef<string | undefined>(undefined);
+  const [selectedTimeText, setSelectedTimeText] = React.useState<string | undefined>(undefined);
 
   const selectTime: TimePickerProps['onTimeSelect'] = React.useCallback(
     (e, data) => {
       setSelectedTime(data.selectedTime);
-      selectedTimeTextRef.current = data.selectedTimeText;
+      setSelectedTimeText(data.selectedTimeText);
```

Comment:
thinking about this some more - why don't you just store your `TimeSelectionData` type directly in state. All you're doing is deconstructing that struct and storing the `selectedTime` and `selectedTimeText` properties in separate states - you could just make it one state


```ts
const [selectedTime, setSelectedTime] = React.useState<TimeSelectionData | null>(null);

  const selectTime: TimePickerProps['onTimeSelect'] = React.useCallback(
    (e, data) => {
      setSelectedTime(data);
```

Reply:
Synced selectedTimeText on a call. It's the submitted text from user input. I renamed it to submittedText.